Point[] in
float64 roll
---
Point[] out
